evervault/crypto/client.py

```diff
@@ -1,109 +1,128 @@
-from .key import Key
-from ..errors.evervault_errors import UndefinedDataError, InvalidPublicKeyError
+from ..errors.evervault_errors import UndefinedDataError, InvalidPublicKeyError, MissingTeamEcdhKey, UnknownEncryptType
 from ..datatypes.map import map_header_type
-from cryptography.hazmat.backends import default_backend
-from cryptography.hazmat.primitives import serialization
-from Crypto.Cipher import AES
-from cryptography.hazmat.primitives import hashes
-from cryptography.hazmat.primitives.asymmetric import padding
+from cryptography.hazmat.primitives.asymmetric import ec
+from cryptography.hazmat.primitives.asymmetric.ec import EllipticCurvePublicKey
+from cryptography.hazmat.primitives.serialization import Encoding, PublicFormat
+from cryptography.hazmat.primitives.ciphers.aead import AESGCM
 from secrets import token_bytes
-from base64 import b64encode
-import json
-import uuid
+import base64
 
 BS = 32
 
 
 class Client(object):
     def __init__(self):
-        self.cage_key = None
         self.public_key = None
+        self.team_ecdh_key = None
+        self.generated_ecdh_key = None
+        self.shared_key = None
 
     def encrypt_data(self, fetch, data):
         if data is None:
             raise UndefinedDataError("Data not defined")
         self.__fetch_cage_key(fetch)
+        self.shared_key = self.__derive_shared_key()
+        
+        if self.shared_key is None or type(self.shared_key) != bytes:
+            raise InvalidPublicKeyError("Provided EC compressed point is invalid")        
+        
+        if type(data) == dict or type(data) == list or type(data) == set:
+            return self.__traverse_and_encrypt(data)
+        elif self.__encryptable_data(data):
+            return self.__encrypt_string(data)
+        else:
+            raise UnknownEncryptType("Cannot encrypt unsupported type")
 
-        if self.cage_key is None or type(self.cage_key) != str:
-            raise InvalidPublicKeyError("Provided public key is invalid")
-        if type(data) == dict:
+    def __traverse_and_encrypt(self, data):
+        if type(data) == list:
+            for idx, item in enumerate(data):
+                if not self.__encryptable_data(item):
+                    data[idx] = self.__traverse_and_encrypt(item)
```

Comment:
Does `__encryptable_data` guarantee that the data will be traversable? This function doesn't have an `else` case, so any type that's not encryptable and not traversable will result in a `None` it seems